hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetResourceTest.java

```diff
@@ -667,6 +668,22 @@ public void getInstalledDistributionSetOfTargetIsEmpty() throws Exception {
                 .andExpect(status().isNoContent()).andExpect(content().string(""));
     }
 
+    @Test
+    @Description("Ensures that a target creation with empty name and a controllerId that exceeds the name length limitation is successful and the name gets truncated.")
+    public void createTargetWithEmptyNameAndLongControllerId() throws Exception {
+        final String randomString = RandomStringUtils.randomAlphanumeric(JpaTarget.CONTROLLER_ID_MAX_SIZE);
+
+        final Target target = entityFactory.target().create().controllerId(randomString).build();
+
+        final String targetList = JsonBuilder.targets(Arrays.asList(target), false);
+
+        mvc.perform(post(MgmtRestConstants.TARGET_V1_REQUEST_MAPPING).content(targetList)
+                .contentType(MediaType.APPLICATION_JSON)).andExpect(status().isCreated())
+                .andExpect(jsonPath("[0].controllerId", equalTo(randomString)))
+                .andExpect(jsonPath("[0].name", equalTo(randomString.substring(0,
```

Comment:
Here you are testing the response for proper values in its body, which is fine. Would it be even better to not only check it via the response body, but additionally check the targets name value by calling the target management (so we are sure its not only the response body which looks fine)? This could look like:
`assertThat(targetManagement.getByControllerID(randomString).get().getName()).isEqualTo(randomString.substring(0, Math.min(JpaTarget.CONTROLLER_ID_MAX_SIZE, NamedEntity.NAME_MAX_SIZE)));`

By adding this, you could then also exclude `randomString.substring(0, Math.min(JpaTarget.CONTROLLER_ID_MAX_SIZE, NamedEntity.NAME_MAX_SIZE))` to a variable like `String expectedTargetName`.

Reply:
Implemented